File: myapp/lib/Models/Matiere.dart

class Matiere{
  int Id_Mat;
  String Designation="";
  double poids;
  int Id_UE;


  Matiere(this.Id_Mat, this.Designation, this.poids, this.Id_UE);

  Map toJson() => {
    'Id_Mat':Id_Mat,
    'Designation':Designation,
    'Poids':poids,
    'Id_UE':Id_UE,
  };
}

File: gestion_note/app/Models/Matiere.php
<?php

namespace App\Models;

use App\Models\UE;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Matiere extends Model
{
    use HasFactory;

    protected $table = "matiere";

    protected $primaryKey = 'Id_Mat';
    protected $fillable = ['Designation', 'Poids', 'Id_UE'];

    public function ue()
    {
        return $this->belongsTo(UE::class);
    }

    public function note()
    {
        return $this->hasMany(Note::class);
    }
}
